bin/populate-dependency-track

```diff
@@ -0,0 +1,146 @@
+#!/bin/bash
+
+IFS='' read -d '' -r help_text << EOF
+$0
+
+Populates dependency track with SBOMs for the supplied repositories
+
+USAGE:
+$0 [FLAGS] <ARGUMENTS>...
+
+FLAGS:
+-t <target>: the name of the local fly target to use to submit jobs (**REQUIRED**)
+-c <resources-dir>: path to local repository checked out for ci-concourse-resources
+-d <dir>: path to local repository which contains checked out repos
+
+ARGUMENTS
+* Repos list file - a file containing a list of repos to generate SBOMs for
+  each line contains a different repository name.
+
+EOF
+
+function log() {
+    printf -- '[%s] - %s - %s\n' "$(date +'%FT%T%z')" "${1:?Level required}" "${2:?Message required}"
+}
+
+function clone_repo {
+    local repo_name="${1:?repo_name required}"
+    local branch_name="${2}"
+    log 'DEBUG' "cloning ${repo_name}"
+
+    if command -v gh > /dev/null 2>&1; then
+        if [[ -z "${branch_name}" ]]; then
+            gh repo clone companieshouse/"${repo_name}" "${local_dir}"/"${repo_name}"
+        else
+            gh repo clone companieshouse/"${repo_name}" "${local_dir}"/"${repo_name}" -- -b "${branch_name}"
+        fi
+    else
+        (
+            cd "${local_dir}" || exit 1
+
+            if [[ -z "${branch_name}" ]]; then
+                git clone git@github.com:companieshouse/"${repo_name}".git
+            else
+                git clone -b "${branch_name}" git@github.com:companieshouse/"${repo_name}".git
+            fi
+        )
+    fi
+}
+
+function resolve_version() {
+    local repository_name="$1"
+    (
+        cd "${local_dir}"/"${repository_name}" || exit 1
+
+        gh release view --json name --jq .name || git describe --tags --abbrev=0 || cat version
+    )
+}
+
+function set_parameters() {
+    DEP_TRACK_SERVER='((secrets.dependency-track-server))'
+    DEP_TRACK_API_KEY='((secrets.dependency-track-api-key))'
```

Comment:
Do we have even this pair ?

Not sure if we can reuse any of the following ones:
- [new Concourse](https://github.com/companieshouse/ci-concourse-resources/blob/b2e4c3e5aaac46fa2b9cba3e0ed07181a0dfb36a/tasks/utility/cyclonedx/create-sbom/task.yml#L8-L9)
```
    aws_access_key_id: ((secrets.shared-services-aws-access-key-id))
    aws_secret_access_key: ((secrets.shared-services-aws-secret-access-key))
```
- [old Concourse](https://github.com/companieshouse/ci-concourse-resources/blob/9fb01642e7bb732a8067a6254187d6d81a3e785a/tasks/utility/cyclonedx/create-sbom/task.yml#L8-L9):
```
    aws_access_key_id: ((shared-services-aws-access-key-id))
    aws_secret_access_key: ((shared-services-aws-secret-access-key))
```

Reply:
They're defined by the task - but raises a good point about using new concourse. I'll add a pr to validate the target is correct